### PR TITLE
add option for reproducible-resource-limit

### PR DESCRIPTION
**Pull request by damien-zufferey-sonarsource**

### CODE REVIEW

File: SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/SMTInterpol.java [hunk 1]
```diff
@@ -1331,20 +1349,21 @@ private final boolean getBooleanOption(final String option) {
...
+		mTimeout = new TimeoutHandler(request);
+		mResourceLimit = new ResourceLimit(mTimeout);
 	}
 
 	/**
 	 * Use with caution.
 	 */
 	public TerminationRequest getTerminationRequest() {
-		return mCancel.getTerminationRequest();
+		return mResourceLimit;
```
Comment (jhoenicke) [inline SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/SMTInterpol.java:1367]: I think it should return mCancel, to match setTerminationRequest (and the previous behavior).

**Review by jhoenicke — CHANGES_REQUESTED** [review]
Just some minor changes, see code comments.
Otherwise it looks good and is ready to merge.

File: SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/SMTInterpol.java [hunk 2]
```diff
@@ -1331,20 +1349,21 @@ private final boolean getBooleanOption(final String option) {
...
 	public boolean isTerminationRequested() {
-		return mCancel.isTerminationRequested();
+		return mResourceLimit.isTerminationRequested();
 	}
 
 	/**
 	 * Use with caution.
 	 */
 	public void setTerminationRequest(final TerminationRequest request) {
-		mCancel = new TimeoutHandler(request);
```
[the remark this answers is not in the document]
Comment (damien-zufferey-sonarsource) [reply SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/SMTInterpol.java:1341]: Done

File: SMTInterpolTest/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/APITest.java [hunk 3]
```diff
@@ -348,4 +346,51 @@ public void testUndeclared() {
...
+			solver.assertTerm(solver.term("=", is[j], solver.term("+", is[j-1], solver.numeral(BigInteger.ONE))));
+			Term xp1 = solver.term("=", xs[j], solver.term("+", xs[j-1], solver.numeral(BigInteger.ONE)));
+			Term xm1 = solver.term("=", xs[j], solver.term("-", xs[j-1], solver.numeral(BigInteger.ONE)));
+			solver.assertTerm(solver.term("or", xp1, xm1));
+		}
+		solver.assertTerm(solver.term(">", xs[n-1], is[n-1]));
+	}
+
+	@Test
+	public void testPreproducibleResourceLimit() {
```
[the remark this answers is not in the document]
Comment (damien-zufferey-sonarsource) [reply SMTInterpolTest/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/APITest.java]: Oops, fixed

File: SMTInterpolTest/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/APITest.java [hunk 4]
```diff
@@ -348,4 +346,51 @@ public void testUndeclared() {
...
+			solver.setOption(":reproducible-resource-limit", limits[i]);
+			randomWalk(solver, 10);
+			LBool isSat = solver.checkSat();
+			Assert.assertSame("check-sat with limit " + limits[i], expected[i], isSat);
+			Assert.assertEquals("resource limit is " + limits[i], BigInteger.valueOf(limits[i]), solver.getOption(":reproducible-resource-limit"));
+		}
+	}
+
+	@Test
+	public void testPreproducibleResourceLimitChange() {
```
[the remark this answers is not in the document]
Comment (damien-zufferey-sonarsource) [reply SMTInterpolTest/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/APITest.java]: Fixed

File: SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/SMTInterpol.java (hunk 1 above)
Comment (damien-zufferey-sonarsource) [reply SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/SMTInterpol.java]: I check the setter and getter to much the previous behaviour.

File: SMTInterpolTest/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/APITest.java [hunk 5]
```diff
@@ -369,7 +369,7 @@ public static void randomWalk(SMTInterpol solver, int n) {
 	}
 
 	@Test
-	public void testPreproducibleResourceLimit() {
+	public void testRreproducibleResourceLimit() {
```
Comment (jhoenicke) [inline SMTInterpolTest/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/APITest.java:372]: Still an r too much :) also in the other test.

Comment (damien-zufferey-sonarsource) [reply SMTInterpolTest/src/de/uni_freiburg/informatik/ultimate/smtinterpol/smtlib2/APITest.java]: Should be good this time.
I think I may need another coffee.

**Review by jhoenicke — APPROVED** [review]
Looks good now.  Thanks for adding the feature

**Pull request merged**